rules/DowngradePhp72/NodeAnalyzer/FunctionExistsFunCallAnalyzer.php

```diff
@@ -5,44 +5,18 @@
 namespace Rector\DowngradePhp72\NodeAnalyzer;
 
 use PhpParser\Node\Expr\FuncCall;
-use PhpParser\Node\Stmt\If_;
-use Rector\Core\PhpParser\Node\BetterNodeFinder;
-use Rector\Core\PhpParser\Node\Value\ValueResolver;
-use Rector\NodeNameResolver\NodeNameResolver;
+use PHPStan\Analyser\Scope;
+use Rector\NodeTypeResolver\Node\AttributeKey;
 
 final class FunctionExistsFunCallAnalyzer
 {
-    public function __construct(
-        private readonly BetterNodeFinder $betterNodeFinder,
-        private readonly NodeNameResolver $nodeNameResolver,
-        private readonly ValueResolver $valueResolver
-    ) {
-    }
-
     public function detect(FuncCall $funcCall, string $functionName): bool
     {
-        /** @var If_|null $firstParentIf */
-        $firstParentIf = $this->betterNodeFinder->findParentType($funcCall, If_::class);
-        if (! $firstParentIf instanceof If_) {
-            return false;
-        }
-
-        if (! $firstParentIf->cond instanceof FuncCall) {
-            return false;
-        }
-
-        if (! $this->nodeNameResolver->isName($firstParentIf->cond, 'function_exists')) {
-            return false;
-        }
-
-        /** @var FuncCall $functionExists */
-        $functionExists = $firstParentIf->cond;
-
-        $args = $functionExists->getArgs();
-        if (! isset($args[0])) {
+        $scope = $funcCall->getAttribute(AttributeKey::SCOPE);
+        if (! $scope instanceof Scope) {
             return false;
         }
 
-        return $this->valueResolver->isValue($args[0]->value, $functionName);
+        return $scope->isInFunctionExists($functionName);
```

Comment:
Wow, very nice method :) this class could be now removed and the  `isInFunctionExists()` used in the code instead.

Reply:
https://github.com/rectorphp/rector-downgrade-php/pull/123